### PR TITLE
Wallet Rewards Lookup

### PR DESCRIPTION
**Pull request by alchemistgo87**
Addresses: #812

### CODE REVIEW

File: crates/core/migrations/2022-10-17-074109_create_wallet_total_rewards_table/up.sql [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        then total_reward + new.buyer_reward
+        when wallet_address in
+        (
+            select wallet_address
+            from reward_payouts
+            where wallet_address = new.seller
+        )
+        then total_reward + new.seller_reward
+    end
+    where reward_center_address = new.reward_center;
```
Comment (kespinola) [inline crates/core/migrations/2022-10-17-074109_create_wallet_total_rewards_table/up.sql:35]: Needs to handle creating the record for the buyer or the seller when it doesn't exists

**Review by kespinola — CHANGES_REQUESTED** [review]
need to handle init creation of the records.

File: crates/core/migrations/2022-10-17-074109_create_wallet_total_rewards_table/up.sql [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    language plpgsql
+    as $$
+begin
+    update wallet_total_rewards set total_reward =
+    case 
+        when wallet_address in
+        (
+            select wallet_address
+            from reward_payouts
+            where wallet_address = new.buyer
```
Comment (imabdulbasit) [inline crates/core/migrations/2022-10-17-074109_create_wallet_total_rewards_table/up.sql:24]: I don't see wallet_address column in reward_payouts table

File: crates/core/migrations/2022-10-17-074109_create_wallet_total_rewards_table/up.sql [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    language plpgsql
+    as $$
+begin
+    insert into wallet_total_rewards values (new.buyer, new.reward_center, new.buyer_reward)
+    on conflict (wallet_address, reward_center_address)
+    do update set total_reward = total_reward + new.buyer_reward;
+
+    insert into wallet_total_rewards values (new.seller, new.reward_center, new.seller_reward)
+    on conflict (wallet_address, reward_center_address)
+    do update set total_reward = total_reward + new.seller_reward;
```
Comment (kespinola) [inline crates/core/migrations/2022-10-17-074109_create_wallet_total_rewards_table/up.sql:24]:
```suggestion
    on conflict (wallet_address, reward_center_address)
    do update set total_reward = wallet_total_rewards.total_reward + new.buyer_reward;

    insert into wallet_total_rewards values (new.seller, new.reward_center, new.seller_reward)
    on conflict (wallet_address, reward_center_address)
    do update set total_reward = wallet_total_rewards.total_reward + new.seller_reward;
```

Get ambiguous error when postgres executes the function unless include the table name.

**Review by kespinola — CHANGES_REQUESTED** [review]
looks good just 1 item on scoping column by table name.

**Review by imabdulbasit — APPROVED** [review]

**Review by kespinola — APPROVED** [review]

**Pull request merged**